src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -933,10 +934,35 @@ FixedwingPositionControl::control_position(const math::Vector<2> &curr_pos, cons
 			_att_sp.yaw_body = 0;
 		}
 
+	} else if (_control_mode.flag_control_altitude_enabled
+		   && _control_mode.flag_control_offboard_enabled
+		   && pos_sp_curr.valid) {
+		/* Offboard altitude mode, control on yaw rate with fixed altitude */
+		_control_mode_current = FW_POSCTRL_MODE_OFFBOARD_ALTITUDE;
+
+		_hold_alt = _global_pos.alt;
```

Comment:
This _hold_alt doesn't seem to be used in this case, as the pos_sp_curr.z is actually input to the tecs update. Am I missing it? (see also previous comment)

Reply:
Yeah, you are right, it probably makes more sense to set `_hold_alt` to `pos_sp_curr.z`, so that in the case that control is switched to assisted altitude mode then a similar altitude will be retained.